### PR TITLE
SLING-10364 optionally wait for user input before stopping the server

### PR DESCRIPTION
**Pull request by kwin**

### CODE REVIEW

File: src/main/java/org/apache/sling/maven/feature/launcher/StopMojo.java [hunk 1]
```diff
@@ -37,9 +39,20 @@
...
+    @Parameter(property = "feature-launcher.waitForInput", required = false, defaultValue = "false")
+    protected boolean waitForInput;
+
+    @Component
+    private Prompter prompter;
+
     @Override
     public void execute() throws MojoExecutionException, MojoFailureException {
-        
+        if (waitForInput) {
```
Comment (rombert) [inline src/main/java/org/apache/sling/maven/feature/launcher/StopMojo.java:58]: Can we make this conditional of being executed in an interactive mode? Then the error message for when the prompt fails could be cleared.

**Review by enapps-enorman — APPROVED** [review]
Perhaps the wrong issue number in the title?  Looks like it should be SLING-10364

I tried this out and it does seem to work to keep the server running so the changes are ok for me.

My only concern is that the prompt to hit enter can be difficult to see and may get lost in all the other output.  Perhaps something can be done to make that prompt more visible with whitespace padding around the (more descriptive?) prompt text or a different color or something to make it stand out.

Also, would a configurable timeout be useful here to stop waiting if the user forgets to hit enter for a long time? [lower priority]

File: src/main/java/org/apache/sling/maven/feature/launcher/StopMojo.java (hunk 1 above)
Comment (kwin) [reply src/main/java/org/apache/sling/maven/feature/launcher/StopMojo.java:58]: Done in https://github.com/apache/sling-feature-launcher-maven-plugin/pull/1/commits/f5c6159cccbeecc3eedcd86fb8dffb723ba93ad8

**Comment by kwin** [comment]
I added some colored WARNING before the prompt to make it more visible. Adding a timeout is not that simple and IMHO not necessary, if someone uses the Mojo Parameter (usually via Command Line) she/he is not interested in running into a timeout.
I am gonna fix the commit message JIRA ID once I squash and merge.

**Review by rombert — COMMENTED** [review]
LGTM, thanks

**Pull request merged**